File: redsheet-portal/src/app/database-models/v-nano-scope-to-negotiate-objective-record.ts
import { required, maxLength, range, nested } from '@rx/annotations';

export class vNanoScopeToNegotiateObjectiveRecord {
    constructor(vNanoScopeToNegotiateObjectiveRecord?: vNanoScopeToNegotiateObjectiveRecord )  {
        let properties = [ "buy", "date", "focus", "knowAboutThem", "nanoRelationshipRequireId", "nanoScopeToNegotiateId", "nanoScopeToNegotiateObjectiveId", "opponentName", "projectModuleId", "reason", "valueObjectiveId",];
        for (let property of properties)
            if (vNanoScopeToNegotiateObjectiveRecord && vNanoScopeToNegotiateObjectiveRecord[property])
                this[property] = vNanoScopeToNegotiateObjectiveRecord[property];
    }
 
    @required()
    @maxLength(1000)
	buy : string =   undefined;
 
    @required()
	date : Date =   undefined;
 
    @required()
    @maxLength(1000)
	focus : string =   undefined;
 
    @required()
    @maxLength(1000)
	knowAboutThem : string =   undefined;
 
	nanoRelationshipRequireId : number =   0 ;
 
    @range(1,2147483647)
	nanoScopeToNegotiateId : number =   undefined;
 
    @range(1,2147483647)
	nanoScopeToNegotiateObjectiveId : number =   undefined;
 
    @required()
    @maxLength(1000)
	opponentName : string =   undefined;
 
    @range(1,2147483647)
	projectModuleId : number =   undefined;
 
    @required()
    @maxLength(1000)
	reason : string =   undefined;
 
    @range(1,2147483647)
	valueObjectiveId : number =   undefined;


}

File: RedSheet.DbEntities/Models/vNanoScopeToNegotiateObjectiveRecord.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("vNanoScopeToNegotiateObjectiveRecords", Schema = "dbo")]
	[Serializable]
	public partial class vNanoScopeToNegotiateObjectiveRecord  
	{
 
	[Required] 
	[MaxLength(1000)]
	public string Buy { get; set; } 
 
	[Required]
	public System.DateTime Date { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Focus { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string KnowAboutThem { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.None)] 
	[Key]
	public int NanoRelationshipRequireId { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int NanoScopeToNegotiateId { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Range(1, int.MaxValue)]
	public int NanoScopeToNegotiateObjectiveId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string OpponentName { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ProjectModuleId { get; set; } 
 
	[Required] 
	[MaxLength(1000)]
	public string Reason { get; set; } 
 
	[Range(1, int.MaxValue)]
	public int ValueObjectiveId { get; set; } 

	public vNanoScopeToNegotiateObjectiveRecord()
	{
    }

    }
}

